### DISCUSSION TITLE
Question about style selector when selecting template

### DISCUSSION THREAD

**Discussion by Lilyltt** [AvaloniaUI/Avalonia#17033] [Q&A] [answered]
I need to change the boxsize of the checkbox, but I found a problem
![image](https://github.com/user-attachments/assets/4f8fb615-c3e6-47ed-b276-7b4e082b6447)
If I set the selector using the first way, I can set properties normally, but if I set the selector using the second way, nothing will happen.
Is this intentional design or is it a bug? Thanks for your help.
(By the way,I reproduced this style using codebehind, and this also happened.)

**Comment by stevemonaco** [comment, marked as answer]
There's no bug with selectors. It's a current design limitation regarding priority. `CheckBox:unchecked` is a conditional activator, so it promotes the priority from `Style` to `StyleTrigger`. You can check https://docs.avaloniaui.net/docs/guides/styles-and-resources/setter-precedence for more details.

If we look at the Fluent `CheckBox`'s `ControlTheme`, we can see the style uses hardcoded values instead of resources:

https://github.com/AvaloniaUI/Avalonia/blob/e3c64189a4a47d931338285f157a9c31844bcd78/src/Avalonia.Themes.Fluent/Controls/CheckBox.xaml#L38-L46

These will have `Template` priority which is higher priority than `Style`.

There should probably be a push to update styles so they don't use hardcoded values for anything related to spacing or dimensions if the selector design limitation is not addressed anytime in the near future.

**Reply by Lilyltt** [reply]
I know, Thank you very much for your answer!
